src/Application.cpp

```diff
@@ -1097,7 +1160,43 @@ void Application::s_audioCallback(void* udata, Uint8* stream, int len)
 
 void Application::loadGame()
 {
-  std::string path = util::openFileDialog(g_mainWindow, getEmulatorExtensions(_emulator));
+  const struct retro_system_info* info = _core.getSystemInfo();
+  std::string file_types;
+  std::string supported_exts;
+  const char* ext = info->valid_extensions;
+  const char* ptr = ext;
+
+  while (*ptr)
+  {
+    supported_exts += "*.";
+    while (*ptr && *ptr != '|')
+      supported_exts += *ptr++;
+
+    if (!*ptr)
+      break;
+
+    supported_exts += ';';
+    ++ptr;
+  }
```

Comment:
This converts the `info->valid_extensions` to the WinAPI supported extensions list without needing a hardcoded list in Emulator.cpp.  The next lines add the `.zip` extension for cores that we believe support zipped ROMs.

Reply:
The reason I went with hardcoded extensions is that RALibretro does not support some system with some cores, i.e. mGBA is only supported with GBA, despite it emulating GB and GBC. By filtering out `*.gb` and `*.gbc` from the open dialog, we make it harder for users to use mGBA with unsupported systems.

Reply:
I've restored `getEmulatorExtensions`, but it only returns values for cores where the hardcoded list differed from the core-provided list. If it doesn't return a value for a core, the core-provided list will be used.